File: PHP/String.php
<?php

// 連結
$str = "Honda";
$str2 = "Yamaha";
echo $str.$str2."\n";

// 変数展開
echo "{$str}-{$str2}\n";

// 一致判定
$str = "Honda";
$str2 = "Honda";
if (strcmp($str, $str2) == 0) echo "Equal\n";
$str2 = "honda";
if (strcasecmp($str, $str2) == 0) echo "CaseInsensitive Equal\n";

// 長さ
echo strlen($str)."\n";

// 切り出し
$str = "ABCDEFG";
echo substr($str, 2)."\n"; // CDEFG
echo substr($str, 3, 2)."\n"; // DE 4番目から2文字

// 検索
if (strpos($str, "B") !== false) {
    echo "find B\n";
}
if (stripos($str, "b") !== false) {
    echo "find caseInsensitive B\n";
}

// 分割
$str = "A,B,C,D,E,F,G";
$ary = explode(",", $str);
foreach($ary as $s){
    echo $s."\n";
}

// 大文字小文字
echo strtoupper("abcd")."\n";
echo strtolower("ABCD")."\n";

// 置換
$str = "abcabcabc";
echo str_replace('a', 'A', $str)."\n"; // "AbcAbcAbc"

// 前後の空白を削除
$str = " ab cd ";
echo trim($str)."\n";
echo ltrim($str)."\n"; // 先頭のみ除去
echo rtrim($str)."\n"; // 末尾のみ除去

// 文字列と数値
$num = intval("123");
$num = (int)"123";
$str = strval($num);
$str = (string)$num;

?>

File: Perl/String.pl
use utf8;

# 変数展開
my $str = "Japan";
my $str2 = "I am in $str";
print $str2."\n";

# 結合
$str = "ABCD";
$str = $str."EFG";
print $str."\n";

# 置換 パターンマッチ演算子
$str = "ABCABCABC";
$str =~ s/B/X/g; # AXCAXCAXC（すべて置換）
print $str."\n"; 
$str =~ s/C/Z/; # AXZAXCAXC（最初のみ）
print $str."\n";

# 切り出し
$str = "ABCDEFG";
print substr($str, 1, 3)."\n"; # BCD
print substr($str, 1)."\n"; # BCDEFG
print substr($str, -2)."\n"; # FG

# 分割
$str = "A,B,C,D,E,F,G";
my @list = split(/,/, $str);
foreach my $s (@list) {
  print $s."\n";
}

# 検索
$str = "ABCDEFG";
my $pos = index($str, "CD");
print $pos."\n";

# 長さ
my $length = length "ABCDEFG";
print $length."\n";

# フォーマット
$str = sprintf("Temp is %03d", 7); # 007
print $str."\n";

# ヒアドキュメント
my $str = <<EOS;
あいうえお
かきくけこ
さしすせそ
EOS
print $str."\n";